packages/editor-ui/src/components/FixedCollectionParameter.vue

```diff
@@ -1,220 +1,201 @@
-<script lang="ts">
-import { defineComponent } from 'vue';
-import type { PropType } from 'vue';
+<script lang="ts" setup>
 import type { IUpdateInformation } from '@/Interface';
 
 import type { INodeParameters, INodeProperties, INodePropertyCollection } from 'n8n-workflow';
 import { deepCopy, isINodePropertyCollectionList } from 'n8n-workflow';
 
 import { get } from 'lodash-es';
 
-export default defineComponent({
-	name: 'FixedCollectionParameter',
-	props: {
-		nodeValues: {
-			type: Object as PropType<INodeParameters>,
-			required: true,
-		},
-		parameter: {
-			type: Object as PropType<INodeProperties>,
-			required: true,
-		},
-		path: {
-			type: String,
-			required: true,
-		},
-		values: {
-			type: Object as PropType<Record<string, INodeParameters[]>>,
-			default: () => ({}),
-		},
-		isReadOnly: {
-			type: Boolean,
-			default: false,
-		},
-	},
-	data() {
-		return {
-			selectedOption: undefined,
-			mutableValues: {} as Record<string, INodeParameters[]>,
-		};
-	},
-	computed: {
-		getPlaceholderText(): string {
-			const placeholder = this.$locale.nodeText().placeholder(this.parameter, this.path);
-			return placeholder ? placeholder : this.$locale.baseText('fixedCollectionParameter.choose');
-		},
-		getProperties(): INodePropertyCollection[] {
-			const returnProperties = [];
-			let tempProperties;
-			for (const name of this.propertyNames) {
-				tempProperties = this.getOptionProperties(name);
-				if (tempProperties !== undefined) {
-					returnProperties.push(tempProperties);
-				}
-			}
-			return returnProperties;
-		},
-		multipleValues(): boolean {
-			return !!this.parameter.typeOptions?.multipleValues;
-		},
-		parameterOptions(): INodePropertyCollection[] {
-			if (this.multipleValues && isINodePropertyCollectionList(this.parameter.options)) {
-				return this.parameter.options;
-			}
+import { computed, ref, watch, onBeforeMount } from 'vue';
+import { useI18n } from '@/composables/useI18n';
 
-			return (this.parameter.options as INodePropertyCollection[]).filter((option) => {
-				return !this.propertyNames.includes(option.name);
-			});
-		},
-		propertyNames(): string[] {
-			return Object.keys(this.mutableValues || {});
-		},
-		sortable(): boolean {
-			return !!this.parameter.typeOptions?.sortable;
-		},
-	},
-	watch: {
-		values: {
-			handler(newValues: Record<string, INodeParameters[]>) {
-				this.mutableValues = deepCopy(newValues);
-			},
-			deep: true,
-		},
-	},
-	created() {
-		this.mutableValues = deepCopy(this.values);
+const locale = useI18n();
+
+const props = defineProps<{
+	nodeValues: INodeParameters;
+	parameter: INodeProperties;
+	path: string;
+	values: Record<string, INodeParameters[]>;
+	isReadOnly: boolean | undefined;
```

Comment:
values and isReadOnly should be optional (they don't have required=true in defineComponent)